nodejs/week2/src/backend/index.js

```diff
@@ -0,0 +1,102 @@
+const express = require("express");
+const app = express();
+const port = 3000;
+
+///meals///
+/*app.get("/meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  response.send(mealsWithReviews);
+});*/
+
+const mealsRouter = require("./routes/meals.js");
+app.get("/meals", mealsRouter);
+
+const mealsIdRouter = require("./routes/meals.js");
+app.get("/meals/:id", mealsIdRouter);
+
+const mealsMaxPrice = require("./routes/meals.js");
+app.get("/meals", mealsMaxPrice);
+
+const mealsWithMutchTitle = require("./routes/meals.js");
+app.get("/meals", mealsWithMutchTitle);
+
+////cheap-meals///
+/*app.get("/cheap-meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let cheapMealsWithReviews = mealsWithReviews.filter((meal) => meal.price < 70)
+  response.send(cheapMealsWithReviews);
+});*/
+
+const cheapMealRouter = require("./routes/cheap-meals.js");
+app.get("/cheap-meals", cheapMealRouter);
+
+///large-meals///
+/*app.get("/large-meals", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let largeMealsWithReviews = mealsWithReviews.filter((meal) => meal.maxNumberOfGuests > 6);
+  response.send(largeMealsWithReviews);
+});*/
+
+const largeMealRouter = require("./routes/large-meals.js");
+app.get("/large-meals", largeMealRouter);
+
+///meal///
+/*app.get("/meal", (request, response) => {
+  let mealsData = getDataFromJson(meals);
+  let reviewsData = getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let randomMeal = mealsWithReviews[Math.floor(Math.random() * mealsWithReviews.length)]
+  response.send(randomMeal);
+});*/
+const mealRouter = require("./routes/meal.js");
+app.get("/meal", mealRouter);
```

Comment:
You only need these routes in this exercise. You can remove the other ones cheap meals, large meals, etc

| Route                | Description                                                           |
| -------------------- | --------------------------------------------------------------------- |
| `/meals/{id}`        | Respond with the json for the meal with the corresponding `id`        |
| `/meals`             | Respond with the json for all the meals                               |
| `/reservations/{id}` | Respond with the json for the reservation with the corresponding `id` |
| `/reservations`      | Respond with the json for all reservations                            |
| `/reviews/{id}`      | Respond with the json for the review with the corresponding `id`      |
| `/reviews`           | Respond with the json for all reviews                                 |